### ISSUE TITLE
ITV Player 2.5.9 (LITTLEWIZ)

### ISSUE THREAD

**Issue by chrissxuk** [catchup4kodi/catchup4kodi#19]
ITV Player no longer works in Kodi 19. I use this add-on a lot so was wondering if there is any possibility of this being updated?

**Comment by fuzzkat** [comment]
@chrissxuk for your information, any plugins that are written in Python 2 will need rewriting in Python 3 before they will work in Kodi 19.  This is quite an undertaking as Python 3 is a substantially different language.  I could imagine it taking a little while.

@catchup4kodi if you would like some help, is there anything we can do?

**Comment by chrissxuk** [comment]
Is the ITV add-on being maintained by anyone? I read somewhere that the add-on had been abandoned. I thought this add-on was previously maintained by littlewiz.

**Comment by fuzzkat** [comment]
It was working on v18 for catch-up, but perhaps not fully.  I think there are some issues with watching live TV.

**Comment by chrissxuk** [comment]
Yes it is still working on 18 for catch-up but my question really relates to the likelihood of it working on 19. If it's been abandoned then it never will.

**Comment by cliffbuck1** [comment]
it still works on 18 but not on 19 as already stated,but is there any chance somebody could have a look at making it 19 compatible

**Comment by chrissxuk** [comment]
ITV Player has stopped working on Kodi 18 now so the addon is currently dead. Is there any chance of this being fixed or rewritten for Kodi 19?

**Comment by andyjefferson** [comment]
Kodi log with current 2.5.9 addon on Kodi 18.9.0 has

2021-05-01 10:23:05.101 T:140054065981568  NOTICE: VideoPlayer::OpenFile: plugin://plugin.video.itv/?url=https%3A%2F%2Fwww.itv.com%2Fhub%2Fendeavour%2F2a1229a0012&mode=3&name=3.+Prey+-+28%2F04%2F2021+19%3A00PM&iconimage=https%3A%2F%2Fhubimages.itv.com%2Fepisode%2F2_1229_0012%3Fw%3D512%26h%3D288
2021-05-01 10:23:05.112 T:140052503398144  NOTICE: Creating InputStream
2021-05-01 10:23:05.112 T:140052503398144 WARNING: CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.itv/?url=https%3A%2F%2Fwww.itv.com%2Fhub%2Fendeavour%2F2a1229a0012&mode=3&name=3.+Prey+-+28%2F04%2F2021+19%3A00PM&iconimage=https%3A%2F%2Fhubimages.itv.com%2Fepisode%2F2_1229_0012%3Fw%3D512%26h%3D288
2021-05-01 10:23:05.112 T:140052503398144   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.itv/?url=https%3A%2F%2Fwww.itv.com%2Fhub%2Fendeavour%2F2a1229a0012&mode=3&name=3.+Prey+-+28%2F04%2F2021+19%3A00PM&iconimage=https%3A%2F%2Fhubimages.itv.com%2Fepisode%2F2_1229_0012%3Fw%3D512%26h%3D288]
2021-05-01 10:23:05.113 T:140052503398144  NOTICE: CVideoPlayer::OnExit()
2021-05-01 10:23:05.213 T:140054065981568  NOTICE: CVideoPlayer::CloseFile()
2021-05-01 10:23:05.213 T:140054065981568  NOTICE: VideoPlayer: waiting for threads to exit
2021-05-01 10:23:05.213 T:140054065981568  NOTICE: VideoPlayer: finished waiting

**Comment by chrissxuk** [comment]
I don't really understand that log but I assume it just confirms that it's not working. It looks like something has changed with the streams at the front end on Wednesday 28/4. Nothing works from 28/4 onwards but everything prior to the 28/4 still works.

**Comment by chrissxuk** [comment]
Where is a good place to look and learn how to fix these add-ons? I would love to take on a project to learn how to fix this add-on. I have some basic programming knowledge from many years ago but plenty of time to learn.

**Comment by cliffbuck1** [comment]
dont know how much coding knowledge you have but python site for beginners is a good place to start

https://www.pythonforbeginners.com/code-snippets-source-code/python-code-examples

https://www.python.org/

**Comment by chrissxuk** [comment]
Thank you for the link that's perfect for me. I have a little basic programming knowledge from many years ago which may help but for this I think it's best to start at the beginning. I will start my project now.

**Comment by cliffbuck1** [comment]
I hope you get this addon working again it's my Mrs goto

**Comment by chrissxuk** [comment]
> I hope you get this addon working again it's my Mrs goto

Have you tried Catch-Up TV & More for what your Mrs wants? This add-on has catch up from STV so has a lot of the same programs as ITV. There is no problem with episodes from 28/4 either. This add-on is in the official Kodi repository.

**Comment by chrissxuk** [comment]
> I hope you get this addon working again it's my Mrs goto

This add-on is being fixed for Matrix see this thread. https://forum.kodi.tv/showthread.php?tid=152888&page=26
